migen/fhdl/specials.py

```diff
@@ -388,7 +388,7 @@ def gn(e):
             content = ""
             formatter = "{:0" + str(int(memory.width / 4)) + "X}\n"
             for d in memory.init:
-                content += formatter.format(d)
+                content += formatter.format(d).zfill(int(memory.width / 4))
```

Comment:
you'd probably want that to be ceiling division:
```suggestion
                content += formatter.format(d).zfill(-(-memory.width // 4))
```